runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -268,6 +268,32 @@ Java_java_lang_Class_isClassADeclaredClass(JNIEnv *env, jobject jlClass, jobject
 	return result;
 }
 
+jboolean JNICALL
+Java_java_lang_Class_isCircularDeclaringClass(JNIEnv *env, jobject recv)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+
+	J9Class *clazz = J9VM_J9CLASS_FROM_HEAPCLASS(currentThread, J9_JNI_UNWRAP_REFERENCE(recv));
+	J9Class *currentClazz = clazz;
+	J9UTF8 *outerClassName = J9ROMCLASS_OUTERCLASSNAME(currentClazz->romClass);
+	while (NULL != outerClassName) {
+		J9Class *outerClass = vmFuncs->internalFindClassUTF8(currentThread, J9UTF8_DATA(outerClassName),
+								J9UTF8_LENGTH(outerClassName), currentClazz->classLoader, J9_FINDCLASS_FLAG_THROW_ON_FAIL);
```

Comment:
I don't think you want throw on fail here. Also, if this returns NULL, you will crash when fetching the outer class name.

Reply:
Agreed and replaced the code with
```
		if (NULL == outerClass){
			break;
		} else if (clazz == outerClass) {
			result = JNI_TRUE;
			break;
		}
```

Reply:
I still don't think you want to throw here if the class isn't found.

Reply:
OK, nevermind - you can't just continue on, so I suppose an exception is the only way out here.

Reply:
Replaced the flag of `internalFindClassUTF8` with `J9_FINDCLASS_FLAG_EXISTING_ONLY` as an exception is required in our case.